src/Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp

```diff
@@ -0,0 +1,154 @@
+ * u^0_{iab} =& (\delta^k_i - n_i n^k) \Phi_{kab} := P^k_i \Phi_{kab} \\
+ * u^{\pm}_{ab} =& \Pi_{ab} \pm n^i \Phi_{iab} - \gamma_2\psi_{ab}
+ * \f}
+ *
+ * where \f$\psi_{ab}\f$ is the spacetime metric, and, \f$\Pi_{ab}\f$ and
+ * \f$\Phi_{iab}\f$ are evolved generalized harmonic fields introduced by first
+ * derivatives of \f$\psi_{ab}\f$; \f$\gamma_2\f$ is a constraint damping
+ * parameter, and \f$n_k\f$ is the unit normal to the surface.
+ *
+ * One can invert these relations to obtain the evolved fields \f$w\f$ in terms
+ * of the characteristic ones, to find:
+ *
+ * \f{align*}
+ * \psi_{ab} =& u^{\psi}_{ab}, \\
+ * \Pi_{ab} =& \frac{1}{2}(u^{+}_{ab} + u^{-}_{ab}) + \gamma_2 u^{\psi}_{ab}, \\
+ * \Phi_{iab} =& \frac{1}{2}(u^{+}_{ab} - u^{-}_{ab}) n_i + u^0_{iab}.
+ * \f}
+ *
+ * The corresponding characteristic speeds \f$v\f$ are given by
+ * [c.f. Lindblom+ (2005), see text between Eq.(34) and Eq.(35)]:
+ * \f{align*}
+ * v_{\psi} =& -(1 + \gamma_1) n_k N^k \\
+ * v_{0} =& -n_k N^k \\
+ * v_{\pm} =& -n_k N^k \pm N
+ * \f}
+ *
+ * where \f$N, N^k\f$ are the lapse and shift respectively, \f$\gamma_1\f$ is a
+ * constraint damping parameter, and \f$n_k\f$ is the unit normal to the
+ * surface.
+ */
+template <size_t Dim, typename Frame>
+struct EvolvedFieldsFromCharacteristicFieldsCompute : db::ComputeTag {
+  using argument_tags = tmpl::list<
+      Tags::ConstraintGamma2, Tags::UPsi<Dim, Frame>, Tags::UZero<Dim, Frame>,
+      Tags::UPlus<Dim, Frame>, Tags::UMinus<Dim, Frame>,
+      ::Tags::Normalized<::Tags::UnnormalizedFaceNormal<Dim, Frame>>>;
+
+  using volume_tags = tmpl::list<>;
+
+  static typename Tags::EvolvedFieldsFromCharacteristicFields<Dim, Frame>::type
+      function(const Scalar<DataVector>& gamma_2,
+          const tnsr::aa<DataVector, Dim, Frame>& u_psi,
+          const tnsr::iaa<DataVector, Dim, Frame>& u_zero,
+          const tnsr::aa<DataVector, Dim, Frame>& u_plus,
+          const tnsr::aa<DataVector, Dim, Frame>& u_minus,
+          const tnsr::i<DataVector, Dim, Frame>& unit_normal_one_form) noexcept;
+};
+
+template <size_t Dim, typename Frame>
+struct CharacteristicFieldsCompute : db::ComputeTag {
+  using argument_tags = tmpl::list<
+      Tags::ConstraintGamma2, gr::Tags::SpacetimeMetric<Dim, Frame, DataVector>,
+      gr::Tags::InverseSpatialMetric<Dim, Frame, DataVector>,
+      Tags::Pi<Dim, Frame>, Tags::Phi<Dim, Frame>,
+      ::Tags::Normalized<::Tags::UnnormalizedFaceNormal<Dim, Frame>>>;
+
+  using volume_tags = tmpl::list<>;
+
+  static typename Tags::CharacteristicFields<Dim, Frame>::type function(
+      const Scalar<DataVector>& gamma_2,
+      const tnsr::aa<DataVector, Dim, Frame>& spacetime_metric,
+      const tnsr::II<DataVector, Dim, Frame>& inverse_spatial_metric,
+      const tnsr::aa<DataVector, Dim, Frame>& pi,
+      const tnsr::iaa<DataVector, Dim, Frame>& phi,
+      const tnsr::i<DataVector, Dim, Frame>& unit_normal_one_form) noexcept;
+};
+}  // namespace GeneralizedHarmonic
```

Comment:
Above line 154, add `// @}`, closing the documentation group, so the function that computes char fields from evolved fields also has doxygen documentation